src/15utility.js

```diff
@@ -138,6 +138,40 @@ var doubleqq = (utils.doubleqq = function (s) {
 	return s.replace(/'/g, "\\'");
 });
 
+/**
+  Unescape double-quoted strings (for SQL identifiers)
+  Handles escaped backslashes (\\\\), escaped double quotes (\\") and doubled double quotes ("")
+  @param {string} s Source string
+  @return {string} Unescaped string
+  @example
+
+  \\\\ => \\
+  \\" => "
+  "" => "
+  \\\\\\" => \\"
+
+  */
+var reEscape = (function () {
+	var placeholder = '\x00ESCAPED_BACKSLASH\x00';
+	return {
+		escapedBackslash: /(\\\\)/g,
+		escapedQuote: /(\\\")/g,
+		doubledQuote: /(\"\")/g,
+		placeholder: placeholder,
+		placeholderRestore: new RegExp(placeholder.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'), 'g'),
```

Comment:
I dont understand this. Why are we replacing things in the placeholder? Why are we making a dynamic new RegExp instead of just making the regex?

Reply:
making a replacement is the wrong approach

The solution is to do replacements that takes into account the potentially prepositioned escaped backslashes, something more along the lines of 

    /(\\\\)*\\(["'])/

and replace that with `$1$2` (or what needs to happen)